sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -1320,35 +1320,41 @@ static bool _has_global_addr(gnrc_netif_t *netif)
     return has_global;
 }
 
+static void _netif_bus_attach_and_subscribe(gnrc_netif_t *netif, msg_bus_entry_t *sub)
+{
+    msg_bus_t *bus = gnrc_netif_get_bus(netif, GNRC_NETIF_BUS_IPV6);
+    msg_bus_attach(bus, sub);
+    msg_bus_subscribe(sub, GNRC_IPV6_EVENT_ADDR_VALID);
+}
```

Comment:
```suggestion
static void _netif_bus_attach_and_subscribe_addr_valid(gnrc_netif_t *netif, msg_bus_entry_t *sub)
{
    msg_bus_t *bus = gnrc_netif_get_bus(netif, GNRC_NETIF_BUS_IPV6);
    msg_bus_attach(bus, sub);
    msg_bus_subscribe(sub, GNRC_IPV6_EVENT_ADDR_VALID);
}
```

to reflect that you subscribe to the `GNRC_IPV6_EVENT_ADDR_VALID`?

Reply:
Alternatively, you could also provide the event-type as a parameter, to make this function more re-usable.

Reply:
Can you please provide your reasoning for going for my first suggestion? After thinking about it, I think I like my alternative more.

Reply:
less code on the caller's side and there is only a single event type implemented currently.

Reply:
Not sure, I would call one more parameter "more code", but ok.